core/src/main/scala/scalafix/rewrite/RewriteCtx.scala

```diff
@@ -1,11 +1,29 @@
 package scalafix.rewrite
+import scala.meta.Tree
+import scala.meta.tokens.Tokens
 import scalafix.ScalafixConfig
 import scalafix.util.AssociatedComments
 import scalafix.util.TokenList
 
 case class RewriteCtx(
     config: ScalafixConfig,
+    tokens: Tokens,
     tokenList: TokenList,
     comments: AssociatedComments,
     semantic: Option[SemanticApi]
 )
+
+object RewriteCtx {
+  def fromCode(ast: Tree,
+               config: ScalafixConfig = ScalafixConfig(),
+               semanticApi: Option[SemanticApi] = None): RewriteCtx = {
```

Comment:
What happens if no SemanticApi is supplied?  Is a new one created, or is functionality diminished in some way?